fleetspeak/src/client/system_service.go

```diff
@@ -78,6 +78,12 @@ func (s *systemService) Start(sc service.Context) error {
 
 // ProcessMessage implements Service.
 func (s *systemService) ProcessMessage(_ context.Context, m *fspb.Message) error {
+	if m.MessageType == "RekeyRequest" {
+		if err := s.client.config.Rekey(); err != nil {
+			// Very unlikely.
+			return fmt.Errorf("unable to rekey client: %v", err)
+		}
```

Comment:
return here if the Rekey() is successful.

Reply:
done